test/e2e/mount_test.go

```diff
@@ -289,5 +320,97 @@ func TestMountSyncSecret(t *testing.T) {
 }
 
 func TestMountRotateSecret(t *testing.T) {
-	t.Skip("TODO: test rotate")
+	secretA := []byte("secreta")
+	secretB := []byte("secretb")
+
+	// Enable rotation.
+	check(execCmd(exec.Command("enable-rotation.sh", f.kubeconfigFile)))
+
+	// Wait for deployment to finish.
+	time.Sleep(45 * time.Second)
+
+	// Create test secret.
+	secretFileA := filepath.Join(f.tempDir, "secretValue-A")
+	check(ioutil.WriteFile(secretFileA, secretA, 0644))
+	check(execCmd(exec.Command(
+		"gcloud", "secrets", "create", f.testRotateSecretID,
+		"--replication-policy", "automatic",
+		"--data-file", secretFileA,
+		"--project", f.testProjectID,
+	)))
+
+	// Deploy the test pod.
+	podFile := filepath.Join(f.tempDir, "test-rotate.yaml")
+	if err := replaceTemplate("templates/test-rotate.yaml.tmpl", podFile); err != nil {
+		t.Fatalf("Error replacing pod template: %v", err)
+	}
+
+	if err := execCmd(exec.Command("kubectl", "apply", "--kubeconfig", f.kubeconfigFile,
+		"--namespace", "default", "-f", podFile)); err != nil {
+		t.Fatalf("Error creating job: %v", err)
+	}
+
+	// As a workaround for https://github.com/kubernetes/kubernetes/issues/83242, we sleep to
+	// ensure that the job resources exists before attempting to wait for it.
+	time.Sleep(5 * time.Second)
+	if err := execCmd(exec.Command(
+		"kubectl", "wait", "pod/test-secret-mounter-rotate",
+		"--for=condition=Ready",
+		"--kubeconfig", f.kubeconfigFile,
+		"--namespace", "default",
+		"--timeout", "5m",
+	)); err != nil {
+		t.Fatalf("Error waiting for job: %v", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	command := exec.Command(
+		"kubectl", "exec", "test-secret-mounter-rotate",
+		"--kubeconfig", f.kubeconfigFile,
+		"--namespace", "default",
+		"--",
+		"cat", "/var/gcp-test-secrets/rotate")
+	command.Stdout = &stdout
+	command.Stderr = &stderr
+	if err := command.Run(); err != nil {
+		fmt.Println("Stdout:", stdout.String())
+		fmt.Println("Stderr:", stderr.String())
+		t.Fatalf("Could not read secret from container: %v", err)
+	}
+	if !bytes.Equal(stdout.Bytes(), secretA) {
+		t.Fatalf("Secret value is %x, want: %x", stdout, secretA)
+	}
+
+	// Rotate the secret.
+	secretFileB := filepath.Join(f.tempDir, "secretValue-B")
+	check(ioutil.WriteFile(secretFileB, secretB, 0644))
+	check(execCmd(exec.Command(
+		"gcloud", "secrets", "versions", "add", f.testRotateSecretID,
+		"--data-file", secretFileB,
+		"--project", f.testProjectID,
+	)))
+
+	// Wait for update.
+	time.Sleep(30 * time.Second)
```

Comment:
Can we add a comment about the frequency of the update?

Reply:
Updated